### PR TITLE
Disable IAST in appsec_runtime_activation scenario

### PR DESCRIPTION
**Pull request by jandro996**
## Motivation

- Having IAST enabled can lead to appsec_runtime_activation scenario behaving incorrectly, masking actual issues.
- Example:
  - When running `tests/appsec/rasp/test_lfi.py` and IAST enabled, the test passes incorrectly.
  - This happens due to a limitation in the current Java tracer implementation, which doesn't support hot instrumentation for LFI exploit prevention.
  - With IAST enabled, instrumentation for LFI prevention occurs at startup (shared with RASP), creating a false impression that the functionality works as expected when it does not.

Jira ticket: [APPSEC-55649]

## Changes

- **Disables IAST in appsec_runtime_activation** 

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

[APPSEC-55649]: https://datadoghq.atlassian.net/browse/APPSEC-55649?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jandro996** [comment]
The changes I’m working on break the IAST tests. However, I’d like to use this PR to emphasize the underlying problem.

Possible Solution:
The issue can be addressed without breaking anything by modifying the problematic scenario and disabling IAST specifically for that case.

Questions:
Would it make sense to proceed with this PR as is to highlight the problem (fixing the IAST test with a new scenario for instance)?
Or do you suggest solving it entirely first, even if it temporarily hides the underlying issue?

File: utils/_context/_scenarios/endtoend.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def __init__(
         agent_env=None,
         tracer_sampling_rate=None,
         appsec_enabled=True,
-        iast_enabled=True,
+        iast_enabled=False,
```
Comment (cbeauchesne) [inline utils/_context/_scenarios/endtoend.py:185]: You'll need to set `iast_enabled=True` for scenario that test it (no a lot of them though)

**Comment by jandro996** [comment]
I've changed the approach to keep the default scenario as it is and disable iast in the scenario with issues

Title and description changed

**Review by cbeauchesne — APPROVED** [review]
Failures are not related (https://datadoghq.atlassian.net/browse/APPSEC-56172)

**Pull request merged**